### ISSUE TITLE
Supporting overwrite of the rn-cli.config.js

### ISSUE THREAD

**Issue by jduthon** [electrode-io/electrode-native#1062]
Hello electrode team!

We ran into a problem on our project, we would like to use  [react-native-svg-transformer](https://github.com/kristerkari/react-native-svg-transformer), using the transformer from the `rn-cli.config.js` as recommended.
This works well if we run the app with the classic `react-native run-android`, though it breaks if trying to run `ern run-android`, as the transformer is needed to handle the svg, and without it they are sent off as assets, which is not supported by android.

The reason for that is in `ern-gen-container` in [generateMiniAppsComposite](https://github.com/electrode-io/electrode-native/blob/12345770a5883d3c1379667295d64c065c50ed1d/ern-container-gen/src/generateMiniAppsComposite.ts#L248) where the file is basically recreated locally.

Wondering if this is something we might fix with a cauldron? But I didn't see any docs about that, and AFAIU the runLocalContainerGen will still be ran with a cauldron anyways, so I guess we should be in the same state?

Happy to open a PR for that if we know what would make sense for electrode, on the top of my head we could:
- If `rn-cli.config.js` is part of the folder of the `miniApp`, then we could copy it instead of creating it? That would mean though users with custom `rn-cli.config.js` have to take care of the `sourceExts`, sounds like an ok drawback. Another problem is with multiple miniApps, what to do if several of them are overwriting the `rn-cli.config.js`.... Not sure if there is a good way to handle that
- Having a CLI argument to pass a file as the `rn-cli.config.js`, should fix the multiple miniApps problem and it is a bit more explicit? Though it doesn't work out of the box and you have to be aware of the flag to use it.
- Attempting to merge the different `rn-cli.config.js` (including the default config), but that could be pretty complicated and lead to unexpected results I think.

If you have any other ideas, happy to make a PR with those.

**Comment by belemaire** [comment]
Hi @jduthon 

Tackling this one a little bit late, you probably figured out a solution or workaround on your own, but FYI we will be releasing [base Composite support](https://github.com/electrode-io/electrode-native/blob/master/docs/platform-parts/composite/index.md) (from https://github.com/electrode-io/electrode-native/pull/1108) in 0.31.0 next week. This feature could be a way to solve your issue.

We faced a similar issue, but more having to do with the fact that we needed to add some top level initialization code to the Composite bundle. Some other people might also need some custom top level babel config or metro config, or as in your case rn cli config.

The approaches you suggested to solve this problem were valid. The problem however with copying `rn-cli.config.js` from a MiniApp has to do with potential conflicts in case multiple MiniApps are defining one (coming to your third bullet point about conflict resoluion, which is quite tricky). Also having an option to provide `rn-cli-config.js` explicitely to all commands creating a composite would have been a way, but where are we storing the file so that it is shared by everyone, and also it does not solve for custom `babel.config.js` or `metro.config.js` or custom top level code for the bundle. Adding all these options to the commands to accommodate for everything would be a pain.
